src/main/java/com/google/cloud/tools/maven/CloudSdkAppEngineFactory.java

```diff
@@ -158,7 +158,12 @@ private AppEngineDevServer createDevServerForVersion(
    */
   public AppEngineDevServer devServerRunAsync(
       int startSuccessTimeout, SupportedDevServerVersion version) {
-    File logFile = new File(mojo.mavenProject.getBuild().getDirectory(), "dev_appserver.out");
+    File logFile =
+        new File(
+            mojo.mavenProject.getBuild().getDirectory() + "/dev-appserver-out/dev_appserver.out");
+    if (!logFile.getParentFile().exists() && !logFile.getParentFile().mkdirs()) {
```

Comment:
The dev-appserver-out should probably be defined as itself instead of using the `getParentFile`. Also, it might be clearer to put this check before resolving logFile.